firmware/Platypus.h

```diff
@@ -85,26 +104,90 @@ namespace platypus
   class Sensor : public Configurable
   {
   public:
-    Sensor(int channel);
+    Sensor(int id);
     virtual ~Sensor();
     
     virtual bool set(const char* param, const char* value);
     virtual char *name() = 0;
-    virtual void onSerial();
+    //virtual void onSerial();
     virtual void loop();
 
   protected:
-    // TODO: Change from channel to struct reference?
-    const int channel_;
+    const int id_;
     
   public:
-    static void onSerial_(void *data);
+    //static void onSerial_(void *data);
     static void onLoop_(void *data);
-    virtual void calibrate(int flag);
+    virtual void calibrate(int flag){};
+  };
+
+  class ExternalSensor : public Sensor
+  {
+  public:
+    ExternalSensor(int id, int port);
+    virtual char  *name() = 0;
+
+  protected:
+    const int port_;
+
   };
   
+  class AnalogSensor : public ExternalSensor 
```

Comment:
Not sure. I haven't tested it, just ported old code over to use the new class structure.